File: algo-credit-check/js/creditCheck.js
exports.creditCheck = function(str) {

    let numArr = str.split('').map(Number)
    
    let len = numArr.length-1
    
    for (let i = 0; i < len; i++){
        if (i % 2 === 0){
            numArr[i] *= 2
            if(numArr[i] > 9) {
                numArr[i] -= 9
            }
        }
    }
    let sums = numArr.reduce((total, elem) => {
        return total + elem
    })
    if (sums % 10 === 0) {
        return "The number is valid!"
    } else {
        return "The number is invalid!"
    }
}

File: algo-credit-check/python/credit_check.py
import functools
from re import M
import itertools
import re

def credit_check(str):
    int_list = re.findall('[0-9]', int(str))
    
    # str_list = list(str)
    # map_object = reversed(list(map(int, str_list)))
    # int_list = list(map_object)
    
    for i, item in enumerate(int_list):
        if (i % 2 == 1):
            int_list[i] *= 2
            if int_list[i] > 9:
                int_list[i] -= 9

    int_list.reverse()
    results = functools.reduce(lambda agg, item : agg + item, int_list)
    if results % 10 == 0:
        return "The number is valid!"
    else:
        return "The number is invalid!"
